src/fileInfo.d.ts

```diff
@@ -0,0 +1,16 @@
+export interface FileInfo {
+	/**
+	 * name: The filename
+	 */
+	name: string
+
+	/**
+	 * size: The filesize in bytes
+	 */
+	size: number
+
+	/**
+	 * contents: The contents of the file
+	 */
+	contents: string
```

Comment:
I'm surprised we put the contents into a string, rather than a Buffer, or even a stream.